### ISSUE TITLE
Proposal: support generic proxies next to Tor

### ISSUE THREAD

**Issue by dsbaars** [lightningnetwork/lnd#6845]
Currently, LND allows for Tor connectivity by connecting to the Tor socks proxy and/or Control port.
No other proxies are currently supported, but this would be beneficial to users since it could provide a way to allow for clearnet, [i2p](https://github.com/PurpleI2P/i2pd) and other connectivity like [cjdns](https://github.com/cjdelisle/cjdns).

### Implementation idea

I have never written in go myself before, but it looks like the code from the `tor` package can be used as a basis for SOCKS5 proxies.
It think it is good to start refactoring to a  generic `proxy` package and let the `tor` package build upon that (which is already suggested in the comments in the source files).

### Considerations
- [Bitcoin Core](https://github.com/bitcoin/bitcoin) already supports [Tor](https://github.com/bitcoin/bitcoin/blob/master/doc/tor.md), [I2P](https://github.com/bitcoin/bitcoin/blob/master/doc/i2p.md) and [cjdns](https://github.com/bitcoin/bitcoin/blob/master/doc/cjdns.md) connectivity.
- There are many types of proxies, personally I would prioritise supporting SOCKS5 and HTTP proxies
- I would recommend supporting frequently used [SOCKS proxy schemes](https://urllib3.readthedocs.io/en/stable/reference/contrib/socks.html) to keep configuration simple. It looks like the already used [net.proxy](https://pkg.go.dev/golang.org/x/net@v0.0.0-20220812174116-3211cb980234/proxy) package supports this.
- The behaviour of `tor.skip-proxy-for-clearnet-targets=true` should be used when next to tor, a generic proxy is used.

**Comment by count-null** [comment]
This is incredibly important IMO. I'm hearing from many node operators who are choosing to abandon Tor-only in favor of a clearnet IPv4 or VPN due to negative effects of Tor network congestion affecting their ability to route payments. At a time when online privacy is under attack, any effort to decrease reliance on Tor should be a priority.

**Comment by fvdpol** [comment]
Great proposal/feature request. Looks like PR #5651 is trying to achieve very similar objectives.

**Comment by guggero** [comment]
While I think adding other proxy/connection types is a good idea in general, just adding generic support for proxies in `lnd` doesn't mean they can be used for p2p connections (at least not end to end) automatically. For the wider network to understand I2P for example, the [spec also needs to be adjusted](https://github.com/lightning/bolts/blob/master/07-routing-gossip.md#the-node_announcement-message).

**Comment by litbot-9000** [comment]
Closing as duplicate of #10106 which tracks dual proxy support (clearnet + Tor in parallel). Unique details from this issue have been summarized there.
